pulpcore/cli/rpm/content.py

```diff
@@ -285,10 +297,81 @@ def upload(
     chunk_size: int,
     **kwargs: t.Any,
 ) -> None:
-    """Create a content unit by uploading a file"""
-    body: t.Dict[str, t.Any]
+    """Create a content unit by uploading a file or files"""
     if isinstance(entity_ctx, PulpRpmPackageContext):
-        result = entity_ctx.upload(file=file, chunk_size=chunk_size, **kwargs)
+        directory = kwargs.get("directory")
+        if (not (file or directory)) or (file and directory):
+            raise click.ClickException(
+                _("You must specify one (and only one) of --file or --directory.")
+            )
+        if file:
+            result = entity_ctx.upload(file=file, chunk_size=chunk_size, **kwargs)
+        else:
+            # Create a temp-repo
+            dest_repo_ctx = kwargs["repository"]
+            tmp_repo_ctx = PulpRpmRepositoryContext(pulp_ctx)
+            body: t.Dict[str, t.Any] = {
+                "name": f"uploadtmp_{dest_repo_ctx.entity['name']}_{uuid4()}",
+                "retain_repo_versions": 1,
+                "autopublish": False,
+            }
+            result = tmp_repo_ctx.create(body=body)
+            # Do the deed
+            try:
+                rpms_path = f"{directory}/*.rpm"
+                rpm_names = glob.glob(rpms_path)
```

Comment:
I think, we could determine this list first and decide on whether which upload method we use depending on 0-1-many files.